cmake/test.cmake

```diff
@@ -8,12 +8,15 @@ set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
 FetchContent_MakeAvailable(googletest)
 
 enable_testing()
-FILE(GLOB cpp_tests test/csrc/test_*.cpp)
-
 include(GoogleTest)
-foreach(t ${cpp_tests})
-    get_filename_component(name ${t} NAME_WE)
-    add_executable(${name} ${t})
+
+set(CTEST test/csrc)
+file(GLOB_RECURSE ALL_TESTS ${CTEST}/*.cpp)
+
+foreach(test ${ALL_TESTS})
+    get_filename_component(name ${test} NAME_WE)
+    add_executable(${name} ${test})
+    target_link_libraries(${name} ${PROJECT_NAME} torch)
     target_link_libraries(${name} ${PROJECT_NAME} gtest_main)
```

Comment:
Do we need to repeat `${PROJECT_NAME}`?

Reply:
I don‘t really know. It fails without it.

Reply:
All right :) Could we try to merge them into to one `target_link_libraries`